Library/Homebrew/brew.sh

```diff
@@ -816,6 +806,22 @@ then
   unset HOMEBREW_RUBY_WARNINGS
 fi
 
+# Check for commands that should call `brew update --auto-update` first.
+AUTO_UPDATE_COMMANDS=(
+  install
+  outdated
+  upgrade
+  bump-formula-pr
+  bump-cask-pr
+  bundle
+  release
+)
+if check-array-membership "${HOMEBREW_COMMAND}" "${AUTO_UPDATE_COMMANDS[@]}" ||
+   [[ "${HOMEBREW_COMMAND}" == "tap" && "${HOMEBREW_ARG_COUNT}" -gt 1 ]]
+then
+  export HOMEBREW_AUTO_UPDATE_COMMAND="1"
+fi
+
```

Comment:
The `unset AUTO_UPDATE_COMMANDS` call should be moved over here from line 324.

Reply:
I don't think it's worth unsetting this, we don't unset most variables because they don't end up in Ruby land.